src/app/store/courses/courses.effects.ts

```diff
@@ -0,0 +1,121 @@
+import {Injectable} from '@angular/core';
+import {Actions, createEffect, Effect, ofType} from '@ngrx/effects';
+import {of, tap} from 'rxjs';
+import {map, mergeMap, catchError} from 'rxjs/operators';
+import {CoursesStoreService} from "../../services/courses-store.service";
+import {
+  requestAllCourses,
+  requestAllCoursesFail,
+  requestAllCoursesSuccess,
+  requestCreateCourse, requestCreateCourseFail, requestCreateCourseSuccess,
+  requestDeleteCourse, requestDeleteCourseFail,
+  requestEditCourse,
+  requestEditCourseFail,
+  requestEditCourseSuccess,
+  requestFilteredCourses,
+  requestFilteredCoursesSuccess,
+  requestSingleCourse,
+  requestSingleCourseFail,
+  requestSingleCourseSuccess
+} from "./courses.actions";
+import {Router} from "@angular/router";
+
+@Injectable()
+export class CoursesEffects {
+
+  getAll$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(requestAllCourses),
+      mergeMap(() => this.coursesStoreService.getAll()
+        .pipe(
+          map(courses => {
+              return requestAllCoursesSuccess({courses: courses})
+            }
+          ),
+          catchError(() => of(requestAllCoursesFail()))
+        )
+      )
+    )
+  );
+
+  filteredCourses$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(requestFilteredCourses),
+      mergeMap(value => this.coursesStoreService.searchCourse(value.title)
+        .pipe(
+          map(courses => (requestFilteredCoursesSuccess({
+            courses: courses !== null ? courses : []
+          })))
+        )
+      )
+    )
+  );
+
+  getSpecificCourse$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(requestSingleCourse),
+      mergeMap(value => this.coursesStoreService.getCourse(value.id)
+        .pipe(
+          map(courses => (requestSingleCourseSuccess({
+            courses: courses !== null ? courses : []
+          })))
+        )
+      ),
+      catchError(() => of(requestSingleCourseFail()))
+    )
+  );
+
+  deleteCourse$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(requestDeleteCourse),
+      mergeMap(value => this.coursesStoreService.deleteCourse(value.id)
+        .pipe(
+          map(() => (requestAllCourses))
```

Comment:
```suggestion
          map(() => (requestAllCourses()))
```
Need to call `requestAllCourses`, since it is not an action yet. It is a function